pkg/node/node.go

```diff
@@ -1239,13 +1285,16 @@ func (s *Service) buildInitiatorsArray(initiators []string) []gopowerstore.Initi
 	var portType gopowerstore.InitiatorProtocolTypeEnum
 	if s.useFC {
 		portType = gopowerstore.InitiatorProtocolTypeEnumFC
-	} else {
+	} else if !s.useFC && s.useISCSI {
 		portType = gopowerstore.InitiatorProtocolTypeEnumISCSI
+	} else {
+		portType = gopowerstore.InitiatorProtocolTypeEnumNVME
 	}
+
 	initiatorsReq := make([]gopowerstore.InitiatorCreateModify, len(initiators))
 	for i, iqn := range initiators {
 		iqn := iqn
-		if !s.useFC && s.opts.EnableCHAP {
+		if !s.useFC && s.useISCSI && s.opts.EnableCHAP {
```

Comment:
Here also !s.useFC check is not needed, as CHAP auth is only for iscsi

Reply:
The !s.useFC check was already present, I just added the useISCSI check so maybe first we can confirm that why this check was present. Thanks

Reply:
Yes, check was already present, but seems unnecessary. we can address it later.